internal/handlers/matcher.go

```diff
@@ -0,0 +1,63 @@
+// MIT License
+//
+// Copyright (c) 2019 sparetimecoders
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package handlers
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// overlaps checks if two AMQP binding patterns overlap
+func overlaps(p1, p2 string) bool {
+	if p1 == p2 {
+		return true
+	}
+	r1 := strings.ContainsAny(p1, "#*")
+	r2 := strings.ContainsAny(p2, "#*")
+
+	if r1 {
+		return match(p1, p2)
```

Comment:
What if p1 doesn't match p2 but p2 matches p1? Not sure if that's even possible. That would mean that both are wildcards. p1: user.#.a and p2: user.# would be such a case, right?

Reply:
I think we only should return here if a match is found and not use else-if